### PR TITLE
Support for up to 8 types in the prepare DSL block

### LINKED ISSUE #84: Add more "prepare" overloads
Currently the `prepare` function within `DataLoaderPropertyDSL` only has support up to 1 argument.
We want to add support for multiple just as we have in the other DSLs.

I would like to be able to do something like this:

```
dataProperty<Int, ContactValue?>("mainContact") {
    prepare { current, ctx: Context, type: ContactType ->
        ...
    }
}
```

### PR DESCRIPTION
**Pull request by AlexeySoshin**
Fixes #84 
Took the same approach as other DSLs, by just providing same function with different number of type arguments.

Test is mostly to validate future changes, as current state is validated by the compiler.

### CODE REVIEW

File: kgraphql/src/test/kotlin/com/apurebase/kgraphql/schema/dsl/DataLoaderPropertyDSLTest.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.apurebase.kgraphql.schema.dsl
+
+import com.apurebase.kgraphql.defaultSchema
+import org.junit.Test
```
Comment (jeggy) [inline kgraphql/src/test/kotlin/com/apurebase/kgraphql/schema/dsl/DataLoaderPropertyDSLTest.kt:4]: Small issue here, we are currently using `import org.junit.jupiter.api.Test`.

File: kgraphql/src/test/kotlin/com/apurebase/kgraphql/schema/dsl/DataLoaderPropertyDSLTest.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        defaultSchema {
+            type<Parent> {
+                dataProperty<Element, Parent>("child") {
+                    prepare { parent, a: A, b: B, c: C, d: D, e: E, f: F, g: G, h: H ->
+                        parent + a + b + c + d + e + f + g + h
+                    }
+                }
+            }
+        }
+    }
```
Comment (jeggy) [inline kgraphql/src/test/kotlin/com/apurebase/kgraphql/schema/dsl/DataLoaderPropertyDSLTest.kt:19]: To fully test this, you could update this test to this instead:

```kotlin
@Test
fun `prepare() should support multiple arguments`() {
    val schema = defaultSchema {
        configure{
            executor = Executor.DataLoaderPrepared
        }
        query("parent") {
            resolver { -> Parent() }
        }
        type<Parent> {
            dataProperty<String, Parent>("child") {
                prepare { _, a: String, b: Int, c: String, d: Int, e: String, f: Int, g: String, h: Int ->
                    "$a $b $c $d $e $f $g $h"
                }
                loader { keys ->
                    keys.map { ExecutionResult.Success(Parent(it)) }
                }
            }
        }
    }
    val results = schema.executeBlocking("""
        {
            parent {
                child(a: "A", b: 1, c: "C", d: 2, e: "E", f: 3, g: "G", h: 4) {
                    data
                }
            }
        }
    """.trimIndent()).deserialize()
    results.extract<String>("data/parent/child/data") shouldBeEqualTo "A 1 C 2 E 3 G 4"
}

class Parent(val data: String = "")
```

File: kgraphql/src/test/kotlin/com/apurebase/kgraphql/schema/dsl/DataLoaderPropertyDSLTest.kt (hunk 1 above)
Comment (AlexeySoshin) [reply kgraphql/src/test/kotlin/com/apurebase/kgraphql/schema/dsl/DataLoaderPropertyDSLTest.kt]: Done

**Comment by jeggy** [comment]
This is now available in version 0.13.2 :ok_hand:

**Pull request merged**